### ISSUE TITLE
🐛 `biome.json` JSON schema for 2.1.X requires specifying `options` for rules if `fix` is given

### ISSUE THREAD

**Issue by Bertie690** [biomejs/biome#7217]
### Environment information

```block
CLI:
  Version:                      2.1.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v24.5.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/10.14.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Workspace:
  Open Documents:               0
```

### What happened?

Long story short, I was trying to upgrade our repository's version of Biome from 2.0.0 to 2.1.4.
Unfortunately, it appears some change in the schema broke the schema wrt. `options` objects.

Now, attempting to alter the fix safety of a rule will create spurious errors that "options" parameters are in fact required, _even for rules that verifiably LACK OPTIONS_.

Repro: [biome.jsonc](https://github.com/user-attachments/files/21763656/biome.jsonc)

(I picked an arbitrary one here, literally any other rule with a fix and no options also errors).

A temporary workaround is specifying `options` as an empty object.

### Expected result

Any schema changes should:
1. Not break existing configs without a migrate script
2. Not require specifying options for rules that do not have options simply to please the JSON schema overlords

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Bertie690** [comment]
Here's a screenshot for y'all visual people:

<img width="536" height="226" alt="Image" src="https://github.com/user-attachments/assets/89525bd8-79de-4468-9a55-1ef996e55335" />

**Comment by ematipico** [comment]
This is a known limitation of our schema, and we haven't figured out a fix yet. You can just add `options: {}` to fix it

**Comment by lukepighetti** [comment]
should this stay open since it's still technically a bug? maybe just mark as 'upstream' or 'external' or something?

**Comment by alex-grover** [comment]
this bug seems to exist in the hosted schema at https://biomejs.dev/schemas/2.2.4/schema.json but not the one that's distributed with the npm package

**Comment by ematipico** [comment]
> this bug seems to exist in the hosted schema at https://biomejs.dev/schemas/2.2.4/schema.json but not the one that's distributed with the npm package

I doubt it. They are the same. If you don't get any error it's probably because you're using a package manager that uses symbolic links e.g. `pnpm`, and the editor isn't able to resolve the schema from a symbolic link

**Comment by alex-grover** [comment]
This is what I'm seeing, the left side is https://www.npmjs.com/package/@biomejs/biome/v/2.2.4 and the right side is https://biomejs.dev/schemas/2.2.4/schema.json:

<img width="3024" height="1964" alt="Image" src="https://github.com/user-attachments/assets/55344017-f071-4ec9-8893-4c72074728dd" />

**Comment by ayangweb** [comment]
https://github.com/biomejs/biome/discussions/7865

v2.2.6 still has this problem.
